File: sample-java/src/main/java/com/dylanc/tracker/sample/java/track/ResultThreadNode.java
package com.dylanc.tracker.sample.java.track;

import androidx.annotation.NonNull;

import com.dylanc.tracker.TrackNode;
import com.dylanc.tracker.TrackParams;

/**
 * @author Dylan Cai
 */
public class ResultThreadNode implements TrackNode {

  public String result = null;

  @Override
  public void fillTackParams(@NonNull TrackParams params) {
    if (result != null) {
      params.put("result", result);
    }
  }
}

File: sample-kotlin/src/main/java/com/dylanc/tracker/sample/track/ResultThreadNode.kt
package com.dylanc.tracker.sample.track

import com.dylanc.tracker.TrackNode
import com.dylanc.tracker.TrackParams

/**
 * @author Dylan Cai
 */
class ResultThreadNode : TrackNode {
  var result: String? = null

  override fun fillTackParams(params: TrackParams) {
    result?.let { params.put("result", it) }
  }
}